lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -199,44 +199,26 @@ def verify_access(self, mock_unit, student_should_have_access, expected_error_ty
 
     def test_has_staff_access_to_preview_mode(self):
         """
-        Tests users have right access to content in preview mode.
+        Test that preview mode is only accessible by staff users.
         """
         course_key = self.course.id
-        usage_key = self.course.scope_ids.usage_id
-        chapter = ItemFactory.create(category="chapter", parent_location=self.course.location)
-        overview = CourseOverview.get_from_id(course_key)
-        test_system = get_test_system()
+        CourseEnrollmentFactory(user=self.student, course_id=self.course.id)
 
-        ccx = CcxFactory(course_id=course_key)
-        ccx_locator = CCXLocator.from_course_locator(course_key, ccx.id)
+        for user in [self.global_staff, self.course_staff, self.course_instructor]:
```

Comment:
I think I would not be able to access instance attributes `self.global_staff` ,`self.course_staff`, and `self.course_instructor` outside the instance i.e in `ddt.data`.
Thoughts?